homeassistant/components/light/flux_led.py

```diff
@@ -216,20 +216,16 @@ def turn_on(self, **kwargs):
         if hs_color:
             rgb = color_util.color_hs_to_RGB(*hs_color)
         else:
-            rgb = None
+            rgb = self._bulb.getRgb()
 
-        brightness = kwargs.get(ATTR_BRIGHTNESS)
+        brightness = kwargs.get(ATTR_BRIGHTNESS) or self.brightness
         effect = kwargs.get(ATTR_EFFECT)
         white = kwargs.get(ATTR_WHITE_VALUE)
 
-        # color change only
-        if rgb is not None:
-            self._bulb.setRgb(*tuple(rgb), brightness=self.brightness)
+        self._bulb.setRgb(*tuple(rgb), brightness=brightness)
 
-        # brightness change only
-        elif brightness is not None:
-            (red, green, blue) = self._bulb.getRgb()
-            self._bulb.setRgb(red, green, blue, brightness=brightness)
+        if white is not None:
+            self._bulb.setWarmWhite255(white)
 
         # random color effect
         elif effect == EFFECT_RANDOM:
```

Comment:
Better! However, since `effects` override all other options, move it up as the first conditional so that an unnecessary `setRgb` call doesn't get made. This current code will always make two calls when there is an `effect`.

Also, if `effect` completely overrides brightness and rgb, I think you should fire off a warning if those conflicting parameters are passed in, to ease troubleshooting for users. Not a hard requirement here as the previous code didn't have that, but if you're making changes anyway and feel like it... ;)

Reply:
Updated effect conditionals for early return to prevent unnecessary call to `setRgb`.  Also updated default value handling to support addition of the warning.